### ISSUE TITLE
Selected text cannot be copied and pasted immediately

### ISSUE THREAD

**Issue by MarkStraub** [microsoft/vscode#181860]
Does this issue occur when all extensions are disabled?: Yes

<details>
<summary>VS Code Info:</summary>

- VS Code Version: 1.79.0-insider (user setup)
- Commit: https://github.com/microsoft/vscode/commit/2c73d2651cc525b27fc697b820743098a204ec01
- Date: 2023-05-04T22:11:08.187Z
- Electron: 22.4.8
- Chromium: 108.0.5359.215
- Node.js: 16.17.1
- V8: 10.8.168.25-electron.0
- OS: Windows_NT x64 10.0.19045
- Sandboxed: Yes
</details>

**Steps to Reproduce:**
1. Mark a text to be copied
2. Press Ctrl + C
3. Press Ctrl + V
4. Nothing happens

**Example:**
![vscode-issue](https://user-images.githubusercontent.com/37868502/237017302-a2e4b8d6-70d3-4659-b311-fd39665e1674.gif)

**Temporary fix:**
Press again after the copied text to insert it

Please feel free to let me know if I can help provide any further information to aid troubleshooting.

Thank you in advance for your time.

**Comment by andregreeff** [comment]
I think you might be getting caught out by some strange "feature" behaviour here, not necessarily a "bug".. the last few versions of VSCode seem to be retaining the current text selection on paste.

to use the same HTML snippet as your example:

![VSCode 1 79 0 - retained selection paste issue](https://user-images.githubusercontent.com/126144414/237010986-235f38a4-4ee8-45ed-90ca-4b9379e1cc58.gif)

the first copy/pasta does exactly the same as your example, but with "screencasting mode" enabled so you can see my keyboard input. thereafter, I select _just_ the `Hello World!` text without the newline before it, and paste again a few times.

tl;dr: when you select text, copy it, then paste immediately, VSCode is replacing your current selection with what you copied, and retaining the same selected characters, which makes it look like nothing has changed.

**Comment by MarkStraub** [comment]
In the previous version (1.78.0) this was working properly. It copied the text, pasted it and then deselected the text to insert it multiple times.

![vscode-correct](https://user-images.githubusercontent.com/37868502/237016043-7d0095a7-6f4b-41ae-bed5-80208bbe3294.gif)

This makes more sense which makes me believe it is a bug.

@andregreeff Thanks for the tip with "screencasting mode". I've update it

**Comment by trueliarx** [comment]
Same problem on the latest insider.

**Comment by rdhar** [comment]
I was able to replicate the same erroneous behaviour in v1.79.0-insider (2023-05-04T22:11:08.187Z) but, just as @trueliarx reported, this issue also appears to have been resolved #181847 on the very latest v1.79.0-insider (2023-05-09T05:23:39.976Z).

**Comment by andregreeff** [comment]
@rdhar I didn't even notice the little blue "1" on my settings icon, until after I saw your message..

multiline copy/pasta is working as expected again, thank you.. :)

**Comment by bpasero** [comment]
Adding Matt, who I believe is working on this.
